tekton/ci/interceptors/github/pkg/github/server.go

```diff
@@ -0,0 +1,108 @@
+package github
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/google/go-github/v34/github"
+	pb "github.com/tektoncd/plumbing/tekton/ci/interceptors/github/pkg/proto/v1alpha1/config_go_proto"
+	"github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
+	"google.golang.org/grpc/codes"
+)
+
+type Server struct {
+	client *http.Client
+
+	// Maps event types -> Interceptor handlers.
+	router map[string]Interceptor
+
+	// Webhook Secret
+	webhookSecret []byte
+}
+
```

Comment:
Would be nice to see how much the interceptor SDK can help reduce the boilerplate code here.

Reply:
Absolutely! Once it's ready let me know and I can take a look in a follow up PR. :D